File: string/count-the-number-of-consistent-strings/count-the-number-of-consistent-strings.go
package countthenumberofconsistentstrings

func countConsistentStrings(allowed string, words []string) int {
	lookup := make(map[rune]bool)

	for _, c := range allowed {
		lookup[c] = true
	}
	consistent := len(words)
	for _, w := range words {
		for _, c := range w {
			if !lookup[c] {
				consistent--
				break
			}
		}
	}

	return consistent

}

File: string/count-the-number-of-consistent-strings/count-the-number-of-consistent-strings.ts
function countConsistentStrings(allowed: string, words: string[]): number {
    const lookup = new Set<string>()
    for (let c of allowed) {
        lookup.add(c);
    }
    let consistent = words.length;

    for (let word of words) {
        for (let c of word) {
            if (!lookup.has(c)) {
                consistent--;
                break;
            }
        }
    }

    return consistent;

};
